thorlcr/activities/xmlwrite/thxmlwrite.cpp

```diff
@@ -70,18 +70,11 @@ class CXmlWriteActivityMaster : public CWriteMasterBase
         }
         else
         {
-            StringBuffer supplied(helper->getHeader());
-            size32_t headerLength = supplied.length();
-            if (headerLength == 0 )
-                headerLength = supplied.set(DEFAULTXMLHEADER).newline().length();
-
-            props.setPropInt(FPheaderLength, headerLength);
-
-            supplied.set(helper->getFooter());
-            size32_t footerLength = supplied.length();
-            if (footerLength == 0 )
-                footerLength = supplied.set(DEFAULTXMLFOOTER).newline().length();
-
+            const char *header = helper->getHeader();
+            size32_t headerLen = header ? strlen(header) : strlen(DEFAULTXMLHEADER)+1; // DEFAULTXMLHEADER+'\n' is default output in slave
+            props.setPropInt(FPheaderLength, headerLen);
+            const char *footer = helper->getFooter();
+            size32_t footerLen = footer ? strlen(footer) : strlen(DEFAULTXMLFOOTER)+1; // DEFAULTXMLFOOTER+'\n' is default output in slave
             props.setPropInt(FPfooterLength, footerLength);
```

Comment:
I think this should be 
props.setPropInt(FPfooterLength, footerLen);

Reply:
you're right, and made me wonder how it compiled..
It's because there are completely unused member variables...
I'll fix and delete the unused members.